platform-sdk/swirlds-common/src/main/java/com/swirlds/common/wiring/schedulers/builders/TaskSchedulerConfigOption.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.common.wiring.schedulers.builders;
+
+/**
+ * Various configuration options for a task scheduler. Note that the task scheduler type uses values from
+ * {@link TaskSchedulerType}, and that the unhandled task capacity is represented as an integer value.
+ */
+public enum TaskSchedulerConfigOption {
```

Comment:
Not a requirement, just a suggestion.
I might have been better if all were enums, and based on the enum itself, it might have an argument or not.

Reply:
then when parsing, you check if the enum has arguments or no, and then optionally parse the arguments

Reply:
Are you saying you would have preferred to see syntax like `TYPE(SEQUENTIAL)`?

My personal preference would be to add features/complexity to the parser only as we need it. At the time, only the capacity has an argument, it seems like a premature feature to generalize it so that any could have an argument. Definitely something to consider if we find multiple configuration options in need of arguments.

Would it be ok if this merges as is, and we consider this type of change in follow up PRs?